xchange-stream-okcoin/src/main/java/info/bitrich/xchangestream/okcoin/OkxStreamingService.java

```diff
@@ -0,0 +1,119 @@
+package info.bitrich.xchangestream.okcoin;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import info.bitrich.xchangestream.okcoin.dto.okx.OkxLoginMessage;
+import info.bitrich.xchangestream.okcoin.dto.okx.OkxStreamingAuthenticator;
+import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+import io.reactivex.Completable;
+import io.reactivex.CompletableSource;
+import io.reactivex.Observable;
+import io.reactivex.disposables.Disposable;
+import org.knowm.xchange.ExchangeSpecification;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public class OkxStreamingService extends JsonNettyStreamingService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OkxStreamingService.class);
+    private static final String LOGIN_SIGN_METHOD = "GET";
+    private static final String LOGIN_SIGN_REQUEST_PATH = "/users/self/verify";
+
+    private final Observable<Long> pingPongSrc = Observable.interval(15, 15, TimeUnit.SECONDS);
+
+    private Disposable pingPongSubscription;
+
+    private final ExchangeSpecification xSpec;
+
+    public OkxStreamingService(String apiUrl, ExchangeSpecification exchangeSpecification) {
+        super(apiUrl);
+        this.xSpec = exchangeSpecification;
+    }
+
+    @Override
+    public Completable connect() {
+        Completable conn = super.connect();
+        return conn.andThen(
+            (CompletableSource)
+                (completable) -> {
+                try {
+                    if (pingPongSubscription != null && !pingPongSubscription.isDisposed()) {
+                        pingPongSubscription.dispose();
+                    }
+
+                    pingPongSubscription = pingPongSrc.subscribe(o ->this.sendMessage("ping"));
```

Comment:
does this api use a proper ping message instead of a ping frame? this library is setup to ping with frames already

Reply:
[Okx-docs-v5-websocket-api-connect](https://www.okx.com/docs-v5/en/#websocket-api-connect)

> 2. If the timer is triggered, which means that no new message is received within N seconds, send the String 'ping'.
> 3. Expect a 'pong' as a response. If the response message is not received within N seconds, please raise an error or reconnect.

Only String `ping` and `pong`.